src/javascripts/components/cookie-functions.js

```diff
@@ -246,5 +243,10 @@ function setCookie (name, value, options) {
 
 function deleteCookie (name) {
   document.cookie = name + '=; expires=Thu, 01 Jan 1970 00:00:00 GMT'
-  return null
+
+  // If the cookie still exists, let's try a more thorough way of deleting it,
+  // by specifying the domain and path
+  if (Cookie(name)) {
+    document.cookie = name + '=;expires=Thu, 01 Jan 1970 00:00:00 GMT;domain=' + window.location.hostname.replace(/^www\./, '.') + ';path=/'
+  }
```

Comment:
I'm not sure I understand why this is necessary. Can we add a unit test that covers this scenario?